mindmeld/auto_annotator.py

```diff
@@ -401,27 +416,35 @@ def parse(self, sentence, **kwargs):
 
 
 class SpacyAnnotator(Annotator):
-    """ (English) Annotator class that uses spacy to generate annotations.
-    Supported entities include: "sys_time", "sys_interval", "sys_duration", "sys_number",
-    "sys_amount-of-money", "sys_distance", "sys_weight", "sys_ordinal", "sys_quantity",
-    "sys_percent", "sys_org", "sys_loc", "sys_person", "sys_gpe", "sys_norp", "sys_fac",
-    "sys_product", "sys_event", "sys_law", "sys_langauge", "sys_work-of-art", "sys_other-quantity",
+    """Annotator class that uses spacy to generate annotations.
+    Depending on the language, supported entities can include: "sys_time", "sys_interval",
+    "sys_duration", "sys_number", "sys_amount-of-money", "sys_distance", "sys_weight",
+    "sys_ordinal", "sys_quantity", "sys_percent", "sys_org", "sys_loc", "sys_person",
+    "sys_gpe", "sys_norp", "sys_fac", "sys_product", "sys_event", "sys_law", "sys_langauge",
+    "sys_work-of-art", "sys_other-quantity".
     For more information on the supported entities for the Spacy Annotator check the MindMeld docs.
     """
 
-    def __init__(self, app_path, config=None):
-        """ Initializes an annotator.
+    def __init__(
+        self, app_path, config=None, language=None, locale=None, model_size=None
+    ):
+        """Initializes an annotator.
 
         Args:
             app_path (str): The location of the MindMeld app
             config (dict, optional): A config object to use. This will
                 override the config specified by the app's config.py file.
+            language (str, optional): Language as specified using a 639-1/2 code.
+            locale (str, optional): The locale representing the ISO 639-1 language code and \
+                ISO3166 alpha 2 country code separated by an underscore character.
+            model_size (str, optional): Size of the Spacy model to use. ("sm", "md", or "lg")
         """
         super().__init__(app_path=app_path, config=config)
-
-        self.model = self.config.get("spacy_model", EN_CORE_WEB_LG)
-        self.nlp = SpacyAnnotator._load_model(self.model)
-        self.duckling = DucklingRecognizer.get_instance()
+        self.language = language or self.language
+        self.locale = locale or self.locale
+        self.spacy_model_size = model_size or self.config.get("spacy_model_size", "lg")
+        self.model_name = self._get_spacy_model_name()
+        self.nlp = SpacyAnnotator._load_model(self.model_name)
         self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS = {
```

Comment:
Why do we have this attr when its already defined outside of scope?

Reply:
This attr is currently defined in the SpacyAnnotator's init function since it is only used by the SpacyAnnotator as of now. Should I move this out to constants.py instead?